JavaScript/JavaScriptSDK.Tests/CheckinTests/Context/Session.tests.ts

```diff
@@ -376,5 +466,67 @@ class SessionContextTests extends TestClass {
     private restoreFakeCookie() {
         Microsoft.ApplicationInsights.Util["document"] = this.originalDocument;
     }
+
+    // Make a fake document.cookie property that correctly handles its unique getter/setter functionality.
+    // Document.cookie lets you read the entire set of cookies, but on set acts essentially as +=, with
+    // some logic to overwrite existing cookies with the same name.
+    // Document.cookie also silently removes expired cookies.
+    private generateEmulatedCookieStore() {
+        var cookieStore = {
+            _cookies: <string[]>[]
+        };
+        Object.defineProperty(cookieStore, "cookie", {
```

Comment:
It's hard for me to tell whether emulation approach is better in general for the JavaScript SDK. There are other ways to isolate tests from a dependencies like the `cookie` property as well, like creating a [port](http://blogs.msdn.com/b/ericgu/archive/2014/12/01/unit-test-success-using-ports-adapters-and-simulators.aspx). Subjectively, emulation feels like an overkill in the context of this set of changes.